backend/layers/persistence/persistence.py

```diff
@@ -0,0 +1,75 @@
+from typing import List
+from backend.corpora.common.entities.dataset import Dataset
+from backend.layers.common.entities import CollectionMetadata, CollectionVersion, DatasetArtifact, DatasetMetadata, DatasetStatus
+
+
+class DatabaseProviderInterface:
+        
+    def create_collection(self, owner: str, collection_metadata: CollectionMetadata):
```

Comment:
maybe it should return `CollectionVersion`? similarly for all other other `create*` methods.

Reply:
For the persistence layer, I was thinking to return the id of the newly created entity. That said, if the full entity class can be returned without an additional database call or merge logic, we should take a step further and return it.